### PR TITLE
Added cloud clean command

### PR DESCRIPTION
**Pull request by fortmarek**
Resolves https://github.com/tuist/tuist/issues/5191

### Short description 📝

This PR adds a new command `tuist cloud clean` that adds the same web functionality added [here](https://github.com/tuist/tuist/pull/4957) but surfaced in the CLI.

I have additionally updated `tuist cloud init` to create `Config.swift` if it doesn't exist with the appropriate cloud configuration (instead of just printing it out). That made it easier to write an acceptance test and is useful for users as well.

### How to test the changes locally 🧐

Run `tuist cloud clean` in a cloud-enabled project (you can also use tuist itself).

### Contributor checklist ✅

- [x] The code has been linted using run `./fourier lint tuist --fix`
- [x] The change is tested via unit testing or acceptance testing, or both
- [x] The title of the PR is formulated in a way that is usable as a changelog entry
- [x] In case the PR introduces changes that affect users, the documentation has been updated

### Reviewer checklist ✅

- [x] The code architecture and patterns are consistent with the rest of the codebase
- [x] Reviewer has checked that, if needed, the PR includes the label `changelog:added`, `changelog:fixed`, or `changelog:changed`, and the title is usable as a changelog entry

### CODE REVIEW

File: Sources/TuistKit/Services/Cloud/CloudCleanService.swift [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        configLoader: ConfigLoading = ConfigLoader()
+    ) {
+        self.cloudSessionController = cloudSessionController
+        self.cleanRemoteCacheStorageService = cleanRemoteCacheStorageService
+        self.configLoader = configLoader
+    }
+
+    func clean(
+        path: String?
+    ) async throws {
```
Comment (danyf90) [inline Sources/TuistKit/Services/Cloud/CloudCleanService.swift:50]:
```suggestion
    func clean(path: String?) async throws {
```

File: Sources/TuistKit/Services/Cloud/CloudCleanService.swift [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        try await cleanRemoteCacheStorageService.cleanRemoteCacheStorage(
+            serverURL: cloud.url,
+            projectSlug: cloud.projectId
+        )
+
+        logger.info(
+            """
+            Project was successfully cleaned.
+            """
+        )
```
Comment (danyf90) [inline Sources/TuistKit/Services/Cloud/CloudCleanService.swift:64]:
```suggestion
        logger.info("Project was successfully cleaned.")
```

File: Tests/TuistKitTests/Services/Cloud/CloudCleanServiceTests.swift [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+    override func tearDown() {
+        cloudSessionController = nil
+        cleanRemoteCacheStorageService = nil
+        configLoader = nil
+        subject = nil
+        super.tearDown()
+    }
+
+    func test_cloud_init() async throws {
```
Comment (danyf90) [inline Tests/TuistKitTests/Services/Cloud/CloudCleanServiceTests.swift:37]:
```suggestion
    func test_cloud_clean() async throws {
```

File: Tests/TuistKitTests/Services/Cloud/CloudCleanServiceTests.swift [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+                    url: url,
+                    projectId: "project/slug"
+                )
+            )
+        }
+
+        // When
+        try await subject.clean(
+            path: "/some-path"
+        )
```
Comment (danyf90) [inline Tests/TuistKitTests/Services/Cloud/CloudCleanServiceTests.swift:59]:
```suggestion
        try await subject.clean(path: "/some-path")
```

File: Tests/TuistKitTests/Services/Cloud/CloudCleanServiceTests.swift [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+        // Then
+        XCTAssertEqual(cleanedProjectURL?.absoluteString, url.absoluteString)
+        XCTAssertEqual(cleanedProjectSlug, "project/slug")
+        XCTAssertPrinterOutputContains("""
+        Project was successfully cleaned.
+        """)
```
Comment (danyf90) [inline Tests/TuistKitTests/Services/Cloud/CloudCleanServiceTests.swift:66]:
```suggestion
        XCTAssertPrinterOutputContains("Project was successfully cleaned.")
```

File: Tests/TuistKitTests/Services/Cloud/CloudInitServiceTests.swift [hunk 6]
```diff
@@ -39,12 +48,19 @@ final class CloudInitServiceTests: TuistUnitTestCase {
 
             return "slug"
         }
+        configLoader.loadConfigStub = { _ in
+            Config.test(cloud: nil)
+        }
+        configLoader.locateConfigStub = { _ in
+            AbsolutePath("/some-path")
+        }
```
Comment (danyf90) [inline Tests/TuistKitTests/Services/Cloud/CloudInitServiceTests.swift:56]:
```suggestion
        configLoader.loadConfigStub = { _ in Config.test(cloud: nil) }
        configLoader.locateConfigStub = { _ in AbsolutePath("/some-path") }
```

File: Tests/TuistKitTests/Services/Cloud/CloudInitServiceTests.swift [hunk 7]
```diff
@@ -56,4 +72,57 @@ final class CloudInitServiceTests: TuistUnitTestCase {
...
+        var content: String?
+        configLoader.locateConfigStub = { _ in
+            nil
+        }
+        fileHandler.stubWrite = { stubContent, _, _ in
+            content = stubContent
+        }
+        createProjectService.createProjectStub = { _, _, _ in
+            "slug"
+        }
```
Comment (danyf90) [inline Tests/TuistKitTests/Services/Cloud/CloudInitServiceTests.swift:87]:
```suggestion
        configLoader.locateConfigStub = { _ in nil }
        fileHandler.stubWrite = { stubContent, _, _ in content = stubContent }
        createProjectService.createProjectStub = { _, _, _ in "slug" }
```

File: Tests/TuistKitTests/Services/Cloud/CloudInitServiceTests.swift [hunk 8]
```diff
@@ -56,4 +72,57 @@ final class CloudInitServiceTests: TuistUnitTestCase {
...
+        import ProjectDescription
+
+        let config = Config(
+            cloud: .cloud(projectId: "slug", url: "https://cloud.tuist.io/")
+        )
+
+        """, content)
+        XCTAssertPrinterOutputContains("""
+        Tuist Cloud was successfully initialized.
+        """)
```
Comment (danyf90) [inline Tests/TuistKitTests/Services/Cloud/CloudInitServiceTests.swift:108]:
```suggestion
        XCTAssertPrinterOutputContains("Tuist Cloud was successfully initialized.")
```

File: projects/tuist/features/cloud.feature [hunk 9]
```diff
@@ -5,4 +5,5 @@ Feature: Generate a new project using Tuist (suite 1)
     And I run a local tuist cloud server
     And I have a working directory
     Then I copy the fixture ios_app_with_tests into the working directory
-    Then tuist inits new cloud project
\ No newline at end of file
+    Then tuist inits new cloud project
+    Then tuist cleans the cloud project
```
Comment (danyf90) [inline projects/tuist/features/cloud.feature:9]:
```suggestion
    Then tuist cleans the cloud project

```

**Review by danyf90 — APPROVED** [review]

File: Tests/TuistKitTests/Services/Cloud/CloudInitServiceTests.swift (hunk 7 above)
Comment (fortmarek) [reply Tests/TuistKitTests/Services/Cloud/CloudInitServiceTests.swift]: I can keep the change but I think the multiline versions are more readable

**Comment by fortmarek** [comment]
Thanks @danyf90 for the review and pushing this over the finish line 😁

**Pull request merged**